### ISSUE TITLE
Implement cheapo fuzzy matching to get placesDB records on mistyped queries

### ISSUE THREAD

**Issue by johngruen** [mozilla/universal-search-content#141]
It occurred to be that it might be a relatively easy and useful enhancement... 

RN, returned search results are pretty good at fuzzy matching mis-typed strings. eg:

![screen shot 2015-12-13 at 10 41 51 pm](https://cloud.githubusercontent.com/assets/3323249/11772491/c02271b0-a1ea-11e5-92b5-f999006309a1.png)

If we encounter a case where an input string returns a ton of search suggestions but no results from the places DB, it might be reasonable to run a second query of the places DB against the first returned search suggestion string. In cases where this second query returns no results, the user won't notice a difference, but if the user has mis-entered their input string this would give us a quick and dirty way to use already-existing search result string parsing to provide better results from places. 

![screen_shot_2015-12-13_at_10_41_51_pm](https://cloud.githubusercontent.com/assets/3323249/11772508/f00e0592-a1ea-11e5-89c1-29a2d91d1054.png)

**Comment by nchapman** [comment]
I think doing fuzzy matching sounds great. This strategy for doing it is probably not performant enough to be useful, but I'm sure we could figure out another way to do it.

**Comment by jaredhirsch** [comment]
Fuzzy matching is a great idea. I'm currently working on a draft of a technical roadmap, and I think this idea belongs there, not in a bug.

It might be nice to keep longer-term and short-term (but not current sprint) roadmaps updated as part of the sprint planning meetings. For right now, I've got an early draft of some near-term ideas in an etherpad (I've put fuzzy typo matching in there): https://pad.mocotoolsprod.net/search-technical-roadmap-20151221

Closing, :beers:
